adafruit_requests.py

```diff
@@ -47,11 +47,15 @@ def cast(_t, value):
         """No-op shim for the typing.cast() function which is not available in CircuitPython."""
         return value
 
-
 else:
     from ssl import SSLContext
     from types import ModuleType, TracebackType
-    from typing import Any, Dict, List, Optional, Protocol, Tuple, Type, Union, cast
+    from typing import Any, Dict, List, Optional, Tuple, Type, Union, cast
+
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol
```

Comment:
Make this part of the `else` (indent it one level), since it's type-checking only.